lib/git.zsh

```diff
@@ -12,11 +12,53 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+  local branches
+  local info
+  local deatchedtag
+
+  # If oh-my-zsh.hide-status equals to '1', get out.
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" == "1" ]]; then
+    return 0
+  fi
+
+  # If we are on a folder not tracked by git, get out.
+  if [[ "$(__git_prompt_git rev-parse --git-dir 2> /dev/null)" == "" ]]; then
+    return 0
+  fi
+
+  # If oh-my-zsh.classic-info is set, use classic info output of branch only
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.classic-info 2>/dev/null)" == "1" ]]; then
     ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
     ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
     echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+    return 0
+  fi
+
+  branches=$(__git_prompt_git branch --list -a 2> /dev/null)
+  if [[ "${branches}" == "" && "$?" == "0" ]]; then
```

Comment:
You're asking git for a list of all branches only for this condition. There must be a better way.

Reply:
So far I have not found a better way. Doesn't cost anything, though.

Reply:
Yes it does.